File: 비타알고/DonggyunKim00/1등과 2등.js
const fs = require('fs');

const filePath = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const input = fs
  .readFileSync(filePath, 'utf8')
  .toString()
  .trim()
  .split('')
  .map(Number);

const solution = (input) => {
  // '21','12' 를 저장할 store 선언
  const store = new Set();

  for (let i = 0; i < input.length; i++) {
    // 모두 찾은 경우 미리 for문을 탈출
    if (store.size === 2) break;

    const num = `${input[i]}${input[i + 1]}`;
    if (store.has(num)) continue;
    if (num === '12' || num === '21') {
      store.add(num);
      i += 1;
    }
  }

  return store.size === 2 ? 'YES' : 'NO';
};

console.log(solution(input));

File: 비타알고/JunYoungKr/1등과 2등.py
num = input()

# "12"와 "21"의 첫 출현 위치를 찾음
find_12 = num.find("12")
find_21 = num.find("21")
# print(find_12, find_21)
# 만약 "12"가 먼저 나타난다면, 그 다음에 오는 "21"을 확인
if find_12 != -1 and num.find("21", find_12 + 2) != -1:
    print("Yes")
# 아니면 "21"이 먼저 나타나고, 그 다음에 오는 "12"를 확인
elif find_21 != -1 and num.find("12", find_21 + 2) != -1:
    print("Yes")
else:
    print("No")
